src/audio/pcm_converter/pcm_converter_generic.c

```diff
@@ -130,6 +130,37 @@ static void pcm_convert_s32_to_s24(struct comp_buffer *source,
 
 #endif /* CONFIG_FORMAT_S24LE && CONFIG_FORMAT_S32LE */
 
+#ifdef CONFIG_FORMAT_FLOAT && (CONFIG_FORMAT_S16LE || CONFIG_FORMAT_S24LE || CONFIG_FORMAT_S32LE)
+
+static inline int32_t _pcm_shift_left(int32_t d, int32_t a);
+static int32_t _pcm_convert_f_to_i(int32_t src);
+static int32_t _pcm_convert_i_to_f(int32_t src);
+
+static inline int32_t _pcm_shift_left(int32_t d, int32_t a)
+{
+	return (a >= 0) ? (d << a) : (d >> -a);
+}
+
+static int32_t _pcm_convert_f_to_i(int32_t src)
+{
+	int32_t xi, si, dst;
+	enum {mantisa_mask = (1<<23) - 1};
+
+	xi = ((src >> 23) & 0xFF) - 127; /* exponential */
+	si = ((mantisa_mask + 1) | (src & mantisa_mask)); /* mantisa + 1 */
+	si = (si ^ ( src >> 31 )) + (int)( (unsigned int)src >> 31); /* copy sign to mantisa */
+	dst = _pcm_shift_left(si, xi - 23); /* calculate power */
+	dst += _pcm_shift_left(si, xi - 22) & 1; /* add 0.5 to round correctly */
+	return dst;
+}
+
+static int32_t _pcm_convert_i_to_f(int32_t src)
+{
+	return src;
+}
```

Comment:
Why is this a no-op? Are we really sure the format for fixed point and for floating point are identical in this case? Or what is the reasoning here? (I see an involved function for the inverse conversion)

Reply:
(or is this WIP in the sense that this is only a no-op for now?)

Reply:
second one - WIP

Reply:
What exactly is the fixed point format we're converting to? Maybe I could contribute...